front/src/modules/ui/board/components/BoardColumnMenu.tsx

```diff
@@ -62,6 +72,11 @@ export function BoardColumnMenu({
             title={title}
           />
         )}
+        {openMenu === 'opportunity' && (
+          <RecoilScope>
+            <NewCompanyProgressButton onStartCreateCard={true} />
```

Comment:
It feels surprising to use a "button" here: the component is an entity picker and the logic prevents the button from showing.
I feel we could have an `opportunitiesBoardOptions.entitySelectComponent` instead of `opportunitiesBoardOptions.newCardComponent` we have now.

`entitySelectComponent` would then be a specialised component to deal with the entity selection (`Company` here), and passed as a board option.
I think the rest of the logic we currently have in `NewCompanyProgressButton` (mostly the hotkeys) belongs to the `ui/board` module.

Reply:
On it